src/js/background/context-menu.js

```diff
@@ -0,0 +1,371 @@
+          const title = alias.description ? alias.description : alias.address;
+          const id = "fx-private-relay-use-existing-alias_" + alias.id;
+          
+          
+          data.title = title;
+          data.id = id;
+          data.parentId = opts.parentMenu.id;
+          await browser.menus.create(data, relayContextMenus.utils.onCreatedCallback);
+        }
+        
+        return Promise.resolve(1)
+      }
+
+      await browser.menus.create(data, relayContextMenus.utils.onCreatedCallback);
+
+      return Promise.resolve(1)
+      
+    },
+    remove: async (id) => {
+      await browser.menus.remove(id);
+    }
+  }, 
+  utils: {
+    getAliases: async () => {
+      if (await getServerStoragePref()) {
+        try {
+          return await getAliasesFromServer();
+        } catch (error) {
+          // API Error — Fallback to local storage
+          const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+          return relayAddresses;
+        }
+      }
+
+      // User is not syncing with the server. Use local storage.
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      return relayAddresses;
+      
+    },
+    getGeneratedForHistory: async (website) => {
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      
+      return relayAddresses.some(alias => website === alias.generated_for);
+    },
+    getHostnameFromUrlConstructor: (url) => {
+      const { hostname } = new URL(url);
+      return hostname;
+    },  
+    getMostRecentAliases: (array, domain)=> {
+      // Flipped to match the same order as the dashboard
+      array.reverse();
+
+      // Remove any sites that match the current site (inverse of getSiteSpecificAliases())
+      const filteredAliases = array.filter(alias => alias.generated_for !== domain);
+
+      // Limit to 5
+      return filteredAliases.slice(0, 5);
+    },
+    getSiteSpecificAliases: (array, domain)=> {
+
+      // Flipped to match the same order as the dashboard
+      array.reverse();
+
+      const filteredAliases = array.filter(alias => alias.generated_for === domain);
+
+      // If 5 results for specific domain
+      return filteredAliases.slice(0, 5);
+    },
+    getUserStatus: {
+      canGenerateMoreAliases: async ()=> {
+        const { premium } = await browser.storage.local.get("premium");
+    
+        // Short-circuit if the user is premium.
+        if (premium) return true;
+        
+        const { maxNumAliases } = await browser.storage.local.get("maxNumAliases");
+        const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+
+        return (maxNumAliases - relayAddresses.length) > 0;
+      },
+      getNumberOfAliases: async () => {
+        const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+        return relayAddresses.length;
+      },
+      canUpgradeToPremium: async()=> {
```

Comment:
Should this try to account for what country the user is in? Or we can make that a follow-up later?

Reply:
That was dependent on #217 - resolved that by merging in those changes in 54c5092773dbac5896f992492f0d64621c04efba.